actor/v7action/service_instance_list.go

```diff
@@ -19,15 +18,59 @@ type ServiceInstance struct {
 	UpgradeAvailable    types.OptionalBoolean         `json:"upgrade_available"`
 }
 
-func (actor Actor) GetServiceInstancesForSpace(spaceGUID string) ([]ServiceInstance, Warnings, error) {
+type planDetails struct {
+	plan, offering, broker string
+}
+
+func (actor Actor) GetServiceInstancesForSpace(spaceGUID string, omitApps bool) ([]ServiceInstance, Warnings, error) {
 	instances, included, warnings, err := actor.CloudControllerClient.GetServiceInstances(
 		ccv3.Query{Key: ccv3.SpaceGUIDFilter, Values: []string{spaceGUID}},
 		ccv3.Query{Key: ccv3.FieldsServicePlan, Values: []string{"guid", "name", "relationships.service_offering"}},
 		ccv3.Query{Key: ccv3.FieldsServicePlanServiceOffering, Values: []string{"guid", "name", "relationships.service_broker"}},
 		ccv3.Query{Key: ccv3.FieldsServicePlanServiceOfferingServiceBroker, Values: []string{"guid", "name"}},
 		ccv3.Query{Key: ccv3.OrderBy, Values: []string{"name"}},
 	)
+	if err != nil {
+		return nil, Warnings(warnings), err
+	}
+
+	lastOperation := func(lo resources.LastOperation) string {
+		if lo.Type != "" && lo.State != "" {
+			return fmt.Sprintf("%s %s", lo.Type, lo.State)
+		}
+		return ""
+	}
+
+	planDetailsLookup := actor.buildPlanDetailsLookup(included)
+	var boundAppsLookup map[string][]string
+	if !omitApps {
+		var bindingsWarnings ccv3.Warnings
+		boundAppsLookup, bindingsWarnings, err = actor.buildBoundAppsLookup(instances)
+		warnings = append(warnings, bindingsWarnings...)
+		if err != nil {
+			return nil, Warnings(warnings), err
+		}
+	}
+
+	result := make([]ServiceInstance, len(instances))
+	for i, instance := range instances {
+		names := planDetailsLookup[instance.ServicePlanGUID]
+		result[i] = ServiceInstance{
+			Name:                instance.Name,
+			Type:                instance.Type,
+			UpgradeAvailable:    instance.UpgradeAvailable,
+			ServicePlanName:     names.plan,
+			ServiceOfferingName: names.offering,
+			ServiceBrokerName:   names.broker,
+			BoundApps:           boundAppsLookup[instance.GUID],
+			LastOperation:       lastOperation(instance.LastOperation),
+		}
+	}
 
+	return result, Warnings(warnings), nil
+}
+
+func (actor Actor) buildPlanDetailsLookup(included ccv3.IncludedResources) map[string]planDetails {
```

Comment:
I would have a slight personal preference for this to be a standalone pure function rather than a method, because it does not use `actor`.